components/datetime/src/mock.rs

```diff
@@ -68,6 +71,7 @@ impl MockDateTime {
             hour,
             minute,
             second,
+            fractional_second: FractionalSecond::zero(),
```

Comment:
nitpick: Add `fractional_second` as an argument.  Since we haven't reached version 1 yet, we still have leeway to make API changes.  (it also helps with testing.)